validateType.go.tmpl

```diff
@@ -0,0 +1,36 @@
+{{- define "validateType" -}}
+
+{{- $id := .Id -}}
+{{- $type := .Type -}}
+{{- $collectionType := or .CollectionType ""}}
+
+{{- $disallowedTypes := array "null" -}}
+
+{{- if isMapType $type -}}
+    {{- template "validateType" dict "Id" $id "Type" (mapKeyType $type) "CollectionType" "map key" -}}
+    {{- template "validateType" dict "Id" $id "Type" (mapValueType $type) "CollectionType" "map value" -}}
+
+{{- else if isListType $type -}}
+    {{- template "validateType" dict "Id" (printf "%s[element]" $id) "Type" (listElemType $type) "CollectionType" "list element" -}}
+
+{{- else if isCoreType $type -}}
+    {{- if (has $type $disallowedTypes)}}
+        {{- stderrPrintf "Generator error: '%s' with %s type '%s' not supported for Dart generation. Consider using a more well-defined type. \n" $id $collectionType $type -}}
+        {{- exit 1 -}}
+    {{- end -}}
+
+{{- else if isStructType $type -}}
+    {{/*TODO Investigate: I dont know which struct types dont have fields but this check prevents throws. */}}
+    {{- if hasField $type "Fields" -}}
+        {{- range $_, $field := $type.Fields -}}
+            {{- template "validateType" dict "Id" (printf "%s.%s" $id $field.Name) "Type" $field.Type -}}
+        {{- end -}}
+
+    {{- else -}}
+        {{- stderrPrintf "'%s' struct type %s without fields\n" $id $type -}}
```

Comment:
@pkieltyka  @VojtechVitek I am really struggling here. I would like to be able to limit what core types in the schema are supported for a given language (in this case, Dart), but I am running into an issue where only some `struct` types have `Fields` even though they are definitely types with fields.

For instance, run this code generation for the test schema and you will get `'Complex.user' struct type User without fields`

Other issues arise for enum types and (I think) the `nil` type, which I do not seem to be able to validate.

Any suggestions you have would be really apprecitated.

Reply:
I decided to forego schema validation and support all types in the schema as well as be unopinionated about users using keywords in their schemas, which may result in breakages.